clients/unshaded/src/main/java/tachyon/client/FileOutStream.java

```diff
@@ -187,25 +181,23 @@ public void write(byte[] b, int off, int len) throws IOException {
         int tLen = len;
         int tOff = off;
         while (tLen > 0) {
-          if (mCurrentBlockLeftByte == 0) {
+          if (mCurrentBlockOutStream == null) {
+            getNextBlock();
+          } else if (mCurrentBlockOutStream.getRemainingSpaceByte() == 0) {
```

Comment:
yes, since the new code make sure that the mCurrentBlockOutStream is not null and it doesn't get to the end of the block.
besides the logic in BlockOutStream makes sure that mCurrentBlockOutStream.getRemainingSpaceByte() will never return neagtive value.